undertow/src/main/java/org/wildfly/extension/undertow/deployment/ServletResourceManager.java

```diff
@@ -47,18 +47,22 @@
     private final ResourceManager[] externalOverlays;
     private final boolean explodedDeployment;
 
-    public ServletResourceManager(final VirtualFile resourcesRoot, final Collection<VirtualFile> overlays, boolean explodedDeployment, boolean followSymlink, boolean disableFileWatchService, List<String> externalOverlays) throws IOException {
+    public ServletResourceManager(final VirtualFile resourcesRoot, final Collection<VirtualFile> overlays,
+                                  boolean explodedDeployment, boolean followSymlink, boolean disableFileWatchService,
+                                  List<String> externalOverlays) throws IOException {
         this.explodedDeployment = explodedDeployment;
         Path physicalFile = resourcesRoot.getPhysicalFile().toPath().toRealPath();
-        deploymentResourceManager = new PathResourceManager(physicalFile, TRANSFER_MIN_SIZE, true, followSymlink, !disableFileWatchService);
+        deploymentResourceManager = new PathResourceManager(physicalFile, TRANSFER_MIN_SIZE, true,
+                followSymlink, !disableFileWatchService);
         this.overlays = overlays;
         if(externalOverlays == null) {
             this.externalOverlays = new ResourceManager[0];
         } else {
             this.externalOverlays = new ResourceManager[externalOverlays.size()];
             for (int i = 0; i < externalOverlays.size(); ++i) {
                 String path = externalOverlays.get(i);
-                PathResourceManager pr = new PathResourceManager(Paths.get(path), TRANSFER_MIN_SIZE, true, followSymlink, !disableFileWatchService);
+                PathResourceManager pr = new PathResourceManager(Paths.get(path).toRealPath(), TRANSFER_MIN_SIZE,
```

Comment:
For reviewers: the added .toRealPath() is the meaningful change; the rest is whitespace.